### PR TITLE
Alpha

### PR DESCRIPTION
**Pull request by erikkiralj1**
This PR is the initial implementation of a semantic-release plugin to push files and folders to AWS S3 bucket. Bucket name and directory path are passed into the config, with the ability to remove the root directory of the defined path.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+description: Report a problem with `semantic-release-ecr`
+labels: [bug]
+body:
+    - type: checkboxes
+      id: issue-prerequisite
+      attributes:
+          label: Issues
+          options:
+              - label:
+                    'I have checked [existing issues](https://github.com/RimacTechnology/semantic-release-ecr/issues) and there
```
Comment (vuki656) [inline .github/ISSUE_TEMPLATE/bug_report.yml:11]: Old link

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                    are no issues with the same problem.'
+                required: true
+
+    - type: input
+      attributes:
+          label: '`semantic-release-ecr` Version You Are Using'
```
Comment (vuki656) [inline .github/ISSUE_TEMPLATE/bug_report.yml:17]: Old name

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      validations:
+          required: true
+
+    - type: textarea
+      attributes:
+          label: 'Your Config'
+          description: 'Paste the config you are using with `semantic-release-ecr` from semantic release config file'
```
Comment (vuki656) [inline .github/ISSUE_TEMPLATE/bug_report.yml:24]: Old name

File: .github/ISSUE_TEMPLATE/feature_requrest.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+description: Request a feature request for `semantic-release-ecr`
+labels: [enhancement]
+body:
+    - type: checkboxes
+      id: issue-prerequisite
+      attributes:
+          label: Issues
+          options:
+              - label:
+                    I have checked [existing issues](https://github.com/RimacTechnology/semantic-release-ecr/issues) and there are
```
Comment (vuki656) [inline .github/ISSUE_TEMPLATE/feature_requrest.yml:11]: Old link

File: .github/ISSUE_TEMPLATE/feature_requrest.yml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+name: Feature request
+description: Request a feature request for `semantic-release-ecr`
```
Comment (vuki656) [inline .github/ISSUE_TEMPLATE/feature_requrest.yml:2]: Old name

File: .prettierignore [hunk 6]
```diff
@@ -0,0 +1 @@
+CHANGELOG.md # If your project contains it
```
Comment (vuki656) [inline .prettierignore:1]: No need for that comment

File: package.json [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    ],
+    "scripts": {
+        "build": "tsc",
+        "cm": "cz",
+        "postinstall": "husky install",
+        "lint": "yarn lint:eslint & yarn lint:prettier & yarn lint:package-json & yarn lint:spell",
+        "lint:eslint": "eslint './src' --quiet --cache --cache-strategy content --cache-location '.eslintcache/'",
+        "lint:package-json": "npmPkgJsonLint --configFile ./.packagerc.js .",
+        "lint:prettier": "prettier --loglevel warn --no-editorconfig --no-error-on-unmatched-pattern --check \"./**/*{yaml,yml,json,md}\"",
+        "lint:spell": "cspell --no-progress --no-summary '**'",
```
Comment (vuki656) [inline package.json:25]: Rename `spell` to `cspell` to be consistent

File: package.json [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        "prettier": "^2.7.1",
+        "semantic-release": "^19.0.5",
+        "typescript": "^4.8.4"
+    },
+    "peerDependencies": {
+        "semantic-release": ">=19.0.0"
+    },
+    "packageManager": "yarn@3.2.4",
+    "engines": {
+        "node": ">=16.14",
```
Comment (vuki656) [inline package.json:72]: Can you use node 18?

File: src/aws.ts [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+
+    public async deleteFile(bucket: string, pathToDelete: string) {
+        return new Promise((_resolve, reject) => {
+            const deleteParam = {
+                Bucket: bucket,
+                Key: pathToDelete,
+            }
+
+            this.awsS3.deleteObject(deleteParam, (error) => {
+                // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```
Comment (vuki656) [inline src/aws.ts:51]: Is the error null if the response is valid?

File: src/aws.ts [hunk 10]
```diff
@@ -0,0 +1,120 @@
...
+    public async getExistingFiles(bucket: string) {
+        async function existingFilesKeys(
+            s3: S3,
+            param: ListObjectsV2Request,
+            allKeys: string[] = [],
+        ): Promise<string[]> {
+            return new Promise((resolve, reject) => {
+                s3.listObjectsV2(param, async (error, data) => {
+                    // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+                    if (error) {
```
Comment (vuki656) [inline src/aws.ts:68]: Is the error null if the response is valid?

File: src/aws.ts [hunk 11]
```diff
@@ -0,0 +1,120 @@
...
+    public async uploadFile(bucket: string, key: string, body: ReadStream) {
+        const uploadParams = {
+            Body: body,
+            Bucket: bucket,
+            Key: key,
+        }
+
+        return new Promise((resolve, reject) => {
+            this.awsS3.upload(uploadParams, (error: Error, data: ManagedUpload.SendData) => {
+                // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```
Comment (vuki656) [inline src/aws.ts:111]: Is the error null if the response is valid?

**Review by mario-jerkovic — CHANGES_REQUESTED** [review]

**Review by vuki656 — APPROVED** [review]

**Review by mario-jerkovic — APPROVED** [review]

**Pull request merged**